src/DotNetWorker.Core/Diagnostics/FunctionActivitySource.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections.Concurrent;
+using System.Diagnostics;
+using Microsoft.Azure.Functions.Worker.Logging;
+
+namespace Microsoft.Azure.Functions.Worker.Core.Diagnostics
+{
+    internal static class FunctionActivitySource
+    {
+        private static readonly ActivitySource _activitySource = new("Microsoft.Azure.Functions.Worker");
+        private static readonly string _processId = Process.GetCurrentProcess().Id.ToString();
+        private static readonly ConcurrentDictionary<string, string> _categoryCache = new ConcurrentDictionary<string, string>();
+
+        public static Activity? StartInvoke(FunctionContext context)
+        {
+            var activity = _activitySource.StartActivity("Invoke", ActivityKind.Internal, context.TraceContext.TraceParent);
+
+            if (activity != null)
+            {
+                activity.AddTag(LogConstants.InvocationIdKey, context.InvocationId);
```

Comment:
Should we start adopting OpenTelemetry semantic conventions? https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/faas.md

Although that is an experimental status still. Something I am not a fan of is attribute names like `faas.invoked_name`. I would prefer `faas.invoked.name`.

Reply:
Yes! This is the guidance that I was looking for -- I wasn't sure what to call all of these things. I recall a previous attempt at conventions has been abandoned, so I'm not sure what we should be doing with these "experimental" standards...

I think we need to discuss this before we remove the preview tag on this package.